src/gui.c

```diff
@@ -81,6 +91,27 @@ uzbl_gui_init ()
     } else {
         window_init ();
     }
+
+    uzbl.gui_->im_context = gtk_im_context_simple_new ();
+    gtk_im_context_reset (uzbl.gui_->im_context);
+
+    g_signal_connect (uzbl.gui_->im_context, "commit",
+        G_CALLBACK (uzbl_commit_cb), 0);
+}
+
+static void
+uzbl_commit_cb (
+ GtkIMContext* context,
+ const gchar *str,
+ void* dummy)
+{
+    (void)context;
+    (void)dummy;
+
+    uzbl_events_send (KEY_PRESS, NULL,
+        TYPE_STR, "",
```

Comment:
want to attach the last state to the uzbl object right away then?

sorry. this is going a bit off-topic. I think it's a great idea to always pass around the object. it's just sound software design. But trying to limit user_data to only be the uzblcore is dangerous because everything will need to be a field of uzblcore and some stuff can not easily be a single field like "requested_url" with parallel requests. Some cases might call for small focused structs all containing a pointer to the uzblcore ofc.